optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-client/src/main/java/org/optaplanner/workbench/screens/solver/client/editor/TerminationConfigFormViewImpl.java

```diff
@@ -43,179 +40,25 @@
-            @Override
-            public void onValueChange( ValueChangeEvent<Long> event ) {
-                presenter.onUnimprovedMinutesSpentLimit( event.getValue() );
-            }
-        } );
-        this.unimprovedSpendLimitSeconds = unimprovedSpendLimitSeconds;
-        this.unimprovedSpendLimitSeconds.addValueChangeHandler( new ValueChangeHandler<Long>() {
-            @Override
-            public void onValueChange( ValueChangeEvent<Long> event ) {
-                presenter.onUnimprovedSecondsSpentLimit( event.getValue() );
-            }
-        } );
-
+    public TerminationConfigFormViewImpl( final Tree tree ) {
+        this.tree = tree;
         initWidget( uiBinder.createAndBindUi( this ) );
     }
 
     @Override
-    public void setPresenter( TerminationConfigForm form ) {
-        this.presenter = form;
-    }
-
-    @Override
-    public void setDaysSpentLimit( Long days ) {
-        spentLimitDays.setValue( days );
-    }
-
-    @Override
-    public void setHoursSpentLimit( Long hours ) {
-        spentLimitHours.setValue( hours );
+    public void initTree( TreeItem rootTreeItem ) {
+        this.tree.addItem( rootTreeItem );
     }
 
     @Override
-    public void setMinutesSpentLimit( Long minutes ) {
-        spentLimitMinutes.setValue( minutes );
-    }
-
-    @Override
-    public void setSecondsSpentLimit( Long seconds ) {
-        spentLimitSeconds.setValue( seconds );
-    }
-
-    @Override
-    public void setUnimprovedDaysSpentLimit( Long days ) {
-        unimprovedSpendLimitDays.setValue( days );
-    }
-
-    @Override
-    public void setUnimprovedHoursSpentLimit( Long hours ) {
-        unimprovedSpendLimitHours.setValue( hours );
-    }
-
-    @Override
-    public void setUnimprovedMinutesSpentLimit( Long minutes ) {
-        unimprovedSpendLimitMinutes.setValue( minutes );
-    }
-
-    @Override
-    public void setUnimprovedSecondsSpentLimit( Long seconds ) {
-        unimprovedSpendLimitSeconds.setValue( seconds );
-    }
-
-    @Override
-    public void showSpentLimit( final boolean show ) {
-        useSpentLimit.setValue( show );
-        spentLimitDays.setEnabled( show );
-        spentLimitHours.setEnabled( show );
-        spentLimitMinutes.setEnabled( show );
-        spentLimitSeconds.setEnabled( show );
-    }
-
-    @Override
-    public void showUnimprovedSpentLimit( final boolean show ) {
-        useUnimprovedSpentLimit.setValue( show );
-        unimprovedSpendLimitDays.setEnabled( show );
-        unimprovedSpendLimitHours.setEnabled( show );
-        unimprovedSpendLimitMinutes.setEnabled( show );
-        unimprovedSpendLimitSeconds.setEnabled( show );
-    }
-
-    @UiHandler({"useSpentLimit"})
-    public void onUseSpentLimitChange( ClickEvent event ) {
-        presenter.onUseSpentLimitChange( useSpentLimit.getValue() );
+    public void setPresenter( TerminationConfigForm form ) {
+        this.presenter = form;
```

Comment:
The `presetner` field is now unused, so perhaps it may be removed too (and the `setPresenter()` methods as well).